sound/soc/sof/intel/hda-stream.c

```diff
@@ -185,17 +185,6 @@ hda_dsp_stream_get(struct snd_sof_dev *sdev, int direction)
 			direction == SNDRV_PCM_STREAM_PLAYBACK ?
 			"playback" : "capture");
 
-	/*
-	 * Disable DMI Link L1 entry when capture stream is opened.
-	 * Workaround to address a known issue with host DMA that results
-	 * in xruns during pause/release in capture scenarios.
-	 */
-	if (!IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_ALWAYS_ENABLE_DMI_L1))
-		if (stream && direction == SNDRV_PCM_STREAM_CAPTURE)
-			snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR,
-						HDA_VS_INTEL_EM2,
-						HDA_VS_INTEL_EM2_L1SEN, 0);
-
```

Comment:
can you clarify what difference this makes between disabling L1SEN for capture stream when they are opened, compared to disabling L1SEN when they are triggered?

Reply:
The former one is more powerful, imagine that some capture streams are opened but not actually run(e.g. being paused, or suspended), in these cases, we actually don't need this W/A.

The point here is that let's enable this W/A as less as possible, and do it where it is inevitable only.